addons/mail/static/src/model/model_field.js

```diff
@@ -361,74 +401,29 @@ class ModelField {
         }, options);
     }
 
-    /**
-     * Compute method when this field is related.
-     *
-     * @private
-     * @param {mail.model} record
-     */
-    _computeRelated(record) {
-        const [relationName, relatedFieldName] = this.related.split('.');
-        const Model = record.constructor;
-        const relationField = Model.fields[relationName];
-        if (['one2many', 'many2many'].includes(relationField.relationType)) {
-            const newVal = [];
-            for (const otherRecord of record[relationName]) {
-                const OtherModel = otherRecord.constructor;
-                const otherField = OtherModel.fields[relatedFieldName];
-                const otherValue = otherField.get(otherRecord);
-                if (otherValue) {
-                    if (otherValue instanceof Array) {
-                        // avoid nested array if otherField is x2many too
-                        // TODO IMP task-2261221
-                        for (const v of otherValue) {
-                            newVal.push(v);
-                        }
-                    } else {
-                        newVal.push(otherValue);
-                    }
-                }
-            }
-            if (this.fieldType === 'relation') {
-                return [['replace', newVal]];
-            }
-            return newVal;
-        }
-        const otherRecord = record[relationName];
-        if (otherRecord) {
-            const OtherModel = otherRecord.constructor;
-            const otherField = OtherModel.fields[relatedFieldName];
-            const newVal = otherField.get(otherRecord);
-            if (this.fieldType === 'relation') {
-                if (newVal) {
-                    return [['replace', newVal]];
-                } else {
-                    return [['unlink-all']];
-                }
-            }
-            return newVal;
-        }
-        if (this.fieldType === 'relation') {
-            return [];
-        }
-    }
-
     /**
      * Converts given value to expected format for x2many processing, which is
      * an iterable of records.
      *
      * @private
      * @param {mail.model|mail.model[]} newValue
+     * @param {Object} [param1={}]
+     * @param {boolean} [param1.hasToVerify=true] whether the value has to be
+     *  verified @see `_verifyRelationalValue`
      * @returns {mail.model[]}
      */
-    _setRelationConvertX2ManyValue(newValue) {
+    _convertX2ManyValue(newValue, { hasToVerify = true } = {}) {
```

Comment:
same remark as with `hasToUpdateInverse`: I think it's much easier to read code with well-named positional parameter `options`. It's less of an issue than with `hasToUpdateInverse`, but for consistency it's good.